docs/migrate-to-nativecurlclient.md

```diff
@@ -0,0 +1,255 @@
+# Migrate from `Redmine\Client` to `Redmine\Client\NativeCurlClient`
+
+Since `php-redmine-api` v1.8.0 there is a new native cURL client
+`Redmine\Client\NativeCurlClient`. This guide will help you to migrate your code
+to this client.
+
+## 1. Use new client methods
+
+With the new interface `Redmine\Client\Client` there are now standarized methods
+for all clients. The new `Redmine\Client\NativeCurlClient` and the current `Redmine\Client` implementing this interface.
+
+### api() to getApi()
+
+Search in your code for the usage of `$client->api('issue')` and the magic
+getter like `$client->issue`. Then replace this calls with
+`$client->getApi('issue')`.
+
+```diff
+-$issue = $client->issue->show($issueId);
++$issue = $client->getApi('issue')->show($issueId);
+
+-$client->api('issue')->create($data);
++$client->getApi('issue')->create($data);
+```
+
+### getResponseCode() to getLastResponseStatusCode()
+
+Replace every call for `$client->getResponseCode()` with
+`$client->getLastResponseStatusCode()`.
+
+```diff
+-if ($client->getResponseCode() === 500)
++if ($client->getLastResponseStatusCode() === 500)
+{
+    throw new \Exception('Redmine call failed');
+}
+```
+
+### get() to requestGet()
```

Comment:
```suggestion
### `get()` -> `requestGet()`
```
But is there a reason for doing this? I liked the simple approach of `$client->get()`,  `$client->post()`, `$client->put()` or `$client->delete()`

Reply:
I liked the shorter methods too. But the reason is backward compatibility. `get()` doesn't have typed parameters and returns () `string`, `array` or `false`. For forward compatibility we need new methods which I named `request{Get|Post|...}` in lack of better ideas. `requestGet()` has typed parameters and only returns bool. We could rename them back in v2 but this would destroy our work on the forward compatibility.